kustomize/commands/edit/add/addreplacement.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addReplacementOptions struct {
+	Replacement types.ReplacementField
+}
+
+func newCmdAddReplacement(fSys filesys.FileSystem) *cobra.Command {
+	var o addReplacementOptions
+	o.Replacement.Source = &types.SourceSelector{}
+	o.Replacement.Targets = append(o.Replacement.Targets, &types.TargetSelector{})
+	cmd := &cobra.Command{
+		Use:   "replacement",
+		Short: "add an item to replacement field",
+		Long: `this command will add an item to  replacement field in the kustomization file.
+the item will be:
+	- be either a file, or an inline string
+`,
+		Example: `add replacement --path {filepath}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+			return o.RunAddReplacement(fSys)
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Replacement.Path, "path", "", "Path to the replacement file.")
+	return cmd
+}
+
+// Validate validate add replacement command
+func (o *addReplacementOptions) Validate() error {
+	if o.Replacement.Path == "" {
+		return errors.New("must provide path to add replacement")
+	}
+	return nil
+}
+
+// RunAddReplacement runs addReplacement command
+func (o *addReplacementOptions) RunAddReplacement(fSys filesys.FileSystem) error {
+	mf, err := kustfile.NewKustomizationFile(fSys)
+	if err != nil {
+		return err
+	}
```

Comment:
the lint check is interesting.. it's same as the ``edit add patch`` and it still failed the check

Reply:
This might be because `edit add patch` was created before the linters were enabled.

Reply:
seems like this is the reason. then I need to follow the lint hint to wrap it I guess